cdap-conversion/src/main/java/co/cask/cdap/conversion/app/StreamConversionMapReduce.java

```diff
@@ -0,0 +1,135 @@
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import co.cask.cdap.api.stream.GenericStreamEventData;
+import co.cask.cdap.conversion.avro.Converter;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * MapReduce job that reads events from a stream over a given time interval and writes the events out to
+ * files in avro format, as well as registering those files as partitions in Hive.
+ */
+public class StreamConversionMapReduce extends AbstractMapReduce {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamConversionMapReduce.class);
+  public static final String ADAPTER_PROPERTIES = "adapter.properties";
+  public static final String SCHEMA_KEY = "cdap.stream.conversion.output.schema";
+  public static final String HEADERS_KEY = "cdap.stream.conversion.headers";
+  private String sinkName;
+  private String outputPath;
+  private Long partitionTime;
+
+  @Override
+  public void configure() {
+    setDescription("Job to read a chunk of stream events and write them to a FileSet");
+    setMapperResources(new Resources(512));
+  }
+
+  @Override
+  public void beforeSubmit(MapReduceContext context) throws Exception {
+    Job job = context.getHadoopJob();
+    job.setMapperClass(StreamConversionMapper.class);
+    job.setNumReduceTasks(0);
+    job.setMapOutputKeyClass(AvroKey.class);
+    job.setMapOutputValueClass(NullWritable.class);
+
+    Map<String, String> runtimeArgs = context.getRuntimeArguments();
+    AdapterArguments adapterArguments = new AdapterArguments(runtimeArgs);
+    sinkName = adapterArguments.getSinkName();
+    partitionTime = context.getLogicalStartTime();
+
+    // setup input for the job
+    long endTime = context.getLogicalStartTime();
+    long startTime = endTime - adapterArguments.getFrequency();
+    StreamBatchReadable.useStreamInput(context, adapterArguments.getSourceName(), startTime, endTime,
+                                       adapterArguments.getSourceFormatSpec());
+    context.setMapperResources(adapterArguments.getMapperResources());
+
+    // setup output for the job
+    Map<String, String> sinkArgs = Maps.newHashMap();
+    TimePartitionedFileSetArguments.setOutputPartitionTime(sinkArgs, partitionTime);
+    TimePartitionedFileSet sink = context.getDataset(sinkName, sinkArgs);
+    outputPath = FileSetArguments.getOutputPath(sink.getUnderlyingFileSet().getRuntimeArguments());
```

Comment:
this comes directly from our api, as it is operating on the underlying Fileset that the TimePartitionedFileSet uses